sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -2977,15 +2977,15 @@ class DataSourceV2SQLSuiteV1Filter
         exception = intercept[AnalysisException] {
           sql("SELECT * FROM parquet.`/the/path` VERSION AS OF 1")
         },
-        errorClass = "_LEGACY_ERROR_TEMP_1336",
+        errorClass = "UNSUPPORTED_FEATURE.TIME_TRAVEL",
         sqlState = None,
         parameters = Map("target" -> "path-based tables"))
```

Comment:
Have you considered to introduce sub-classes?

Reply:
My first private version was like this:

<img width="321" alt="image" src="https://github.com/apache/spark/assets/15246973/7cb5d949-7a05-4f4c-a49c-ea91860ddeb9">
But I feel that these subclasses only have different target types for operation, and these types can be extracted, I have my doubts.

Reply:
Don't think all those words are useful for users. How about to show just table/view id like:
```json
      "TIME_TRAVEL" : {
        "message" : [
          "Time travel on the relation: <relationId>."
```

Reply:
Ok, let me do it.